src/Controller/RestApiController.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Www\Controller;
+
+use Www\DataObject\RestResponseDo;
+use Www\Exception\RestException;
+use YapepBase\Config;
+use YapepBase\Controller\HttpController;
+use YapepBase\Exception\ControllerException;
+use YapepBase\Mime\MimeType;
+use YapepBase\Request\IRequest;
+use YapepBase\Response\IResponse;
+use YapepBase\View\RestTemplate;
+
+abstract class RestApiController extends HttpController
+{
+    /** @var RestResponseDo */
+    protected $restResponseDo;
+
+    /** @var string */
+    private $requestBody;
+
+    /** @var array|null */
+    private $requestData;
+
+    public function __construct(IRequest $request, IResponse $response)
+    {
+        parent::__construct($request, $response);
+        $this->restResponseDo = new RestResponseDo();
+    }
+
+    public function run($action)
+    {
+        try {
+            $this->requestBody = file_get_contents('php://input');
+
+            if ($this->requestBody) {
+                $this->requestData = json_decode($this->requestBody, true);
```

Comment:
Maybe it would make sense to check if the decode was successful and return a proper error to the client.
Thinking about using json_last_error() here